forge-game/src/main/java/forge/game/ability/effects/CopySpellAbilityEffect.java

```diff
@@ -128,32 +130,39 @@ public void resolve(SpellAbility sa) {
                             final Player p = (Player) o;
                             if (p.equals(originalTargetPlayer))
                                 continue;
-                            if (p.isValid(type.split(","), chosenSA.getActivatingPlayer(), chosenSA.getHostCard(), sa)) {
+                            if (p.isValid(type.split(","), sa.getActivatingPlayer(), sa.getHostCard(), sa)) {
                                 players.add(p);
                             }
                         }
                     }
-                    valid = CardLists.getValidCards(valid, type, chosenSA.getActivatingPlayer(), chosenSA.getHostCard(), sa);
+                    valid = CardLists.getValidCards(valid, type, sa.getActivatingPlayer(), sa.getHostCard(), sa);
                     Card originalTarget = Iterables.getFirst(getTargetCards(chosenSA), null);
                     valid.remove(originalTarget);
 
                     if (sa.hasParam("ChooseOnlyOne")) {
-                        Card choice = controller.getController().chooseSingleEntityForEffect(valid, sa, Localizer.getInstance().getMessage("lblChooseOne"), null);
+                        FCollectionView<GameEntity> all = new FCollection<>();
```

Comment:
shouldn't the type be just FCollection? so you don't need to cast below?

Reply:
Probably, I'm not very good with `X` vs. `XView` yet 😅